packages/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -186,9 +190,12 @@ export class ConnectionPropsForSessCfg {
             if (ConnectionPropsForSessCfg.propHasValue(finalSessCfg.port) === false) {
                 let answer: any;
                 while (answer === undefined) {
-                    answer = await CliUtils.promptWithTimeout(
-                        "Enter the port number for your service: "
-                    );
+                    const text = "Enter the port number for your service: "
+                    if (parms) {
+                        answer = await parms.response.console.prompt(text);
+                    } else {
+                        answer = await CliUtils.promptWithTimeout(text);
+                    }
```

Comment:
Having a helper function would be nice (to avoid duplication and reduce maintenance overhead) 😋 
Not a required change though. Just being annoying : )

Reply:
not annoying - good point

Reply:
fixed